src/animation/animation_spritesheet.rs

```diff
@@ -71,6 +71,6 @@ fn apply_animation_player_spritesheet(
 
         animation.frame = index;
         let keyframes = animation_clip.keyframes();
-        *texture_atlas_index = keyframes[index]
+        *texture_atlas_index = keyframes.get(index).unwrap();
```

Comment:
I am generally pretty wary of unwraps, especially in libraries. I was looking into the get function and I think we can just always return a usize here?

```rust
pub fn get(&self, index: usize) -> Option<usize> {
    match self {
        Keyframes::KeyframesVec(vec) => Some(vec[index]),
        Keyframes::KeyframesRange(range) => Some(range.start + index),
    }
}
```

Reply:
I wanted to model it based on the get method on vec.
That said, the implementation is not good :D. It should be `vec.get(index)` instead of `vec[index]` and `range.start + index` need to be `< range.end`otherwise return None.

I think we don't need to properly handle the None case in the animation system - the calculated index should always be valid. With that in mind I would switch from unwrap to expect.